src/main/java/org/opensrp/service/EventService.java

```diff
@@ -234,29 +278,178 @@ public synchronized Event processOutOfArea(Event event) {
-			String eventType = event.getEventType().toLowerCase().contains(GROWTH_MONITORING_EVENT.toLowerCase()) ? GROWTH_MONITORING_EVENT
-					: event.getEventType().toLowerCase().contains(VACCINATION_EVENT.toLowerCase()) ? VACCINATION_EVENT : null;
-			if (eventType != null) {
-				Event newEvent = getNewOutOfAreaServiceEvent(event, birthRegEvent, eventType);
-				addEvent(newEvent, birthRegEvent.getProviderId());
+	private void processOutOfAreaRecurringService(Event event, Event birthRegEvent) {
+		//Get and sort previous recurring services related to the current
+		String recurringServiceTypes = event.getDetails().get(RECURRING_SERVICE_TYPES);
+		List<String> outOfCatchmentServices = Arrays
+				.asList(recurringServiceTypes.substring(1, recurringServiceTypes.length() - 1).split(","));
+		List<Event> previousServices = findByBaseEntityAndType(birthRegEvent.getBaseEntityId(), RECURRING_SERVICE);
+		Map<String, List<Event>> matchedRecurringServices = getMarchedRecurringServices(previousServices,
+				outOfCatchmentServices);
+
+		//Create new recurring service event with correct sequence, incrementing on the old recurring service's
+		//Otherwise create first recurring service
+		for (String service : outOfCatchmentServices) {
+
+			Event newEvent = getNewOutOfAreaServiceEvent(event, birthRegEvent, RECURRING_SERVICE);
+			newEvent.setEntityType(RECURRING_SERVICE_UNDERSCORED);
+
+			//Check if there are any existing recurring services or create the first
+			List<Event> events = matchedRecurringServices.get(service);
+			if (!events.isEmpty()) {
+				Event lastRecurringService = events.get(events.size() - 1);
+
+				Obs obsWithValue = getObsWithValue(lastRecurringService);
+				if (obsWithValue != null) {
+					int newVal = Integer.parseInt((String) obsWithValue.getValues().get(0)) + 1;
+					String newSequence = String.valueOf(newVal);
+					String newFormSubmissionField = String.format("%s_%s", service, newSequence);
+					List<Obs> newObsList = updateObs(lastRecurringService, event, newSequence, newFormSubmissionField);
+					newEvent.setObs(newObsList);
+					addEvent(newEvent, birthRegEvent.getProviderId());
+				}
+			} else {
+				createFirstRecurringService(service, newEvent, birthRegEvent);
+			}
+		}
+	}
+
+	private void createFirstRecurringService(String service, Event newEvent, Event birthRegEvent) {
+		String submissionField = String.format("%s_%s", service, 1);
+		List<Object> values = Collections.singletonList(ObsConstants.CODED_FIELD_VALUE);
+		String codedFieldCode;
+		switch (service.trim()) {
+			case RecurringServiceConstants.DEWORMING:
+				codedFieldCode = ObsConstants.DEWORMING_CODED_FIELD_CODE;
+				break;
+			case RecurringServiceConstants.VIT_A:
+				codedFieldCode = ObsConstants.VIT_A_CODED_FIELD_CODE;
+				break;
+			case RecurringServiceConstants.ITN:
+				codedFieldCode = ObsConstants.ITN_CODED_FIELD_CODE;
+				break;
+			default:
+				codedFieldCode = submissionField;
+				break;
```

Comment:
This specific to a certain use case. Why is it that way? I might not have or understand the context of what you are trying to check for.

Reply:
True. This functionality is only specific to EIR projects. The field code are used when creating recurring service event dynamically. There are only 3 recurring services supported at the moment.